[SRVCOM-1922] Add Serverless 1.25.0 release notes & update feature tables (#51075)
openshift/openshift-docs: modules/serverless-rn-1-25-0.adoc

// Module included in the following assemblies
//
// * serverless/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-25-0_{context}"]
= Release notes for Red Hat {ServerlessProductName} 1.25.0

{ServerlessProductName} 1.25.0 is now available. New features, changes, and known issues that pertain to {ServerlessProductName} on {product-title} are included in this topic.

[id="new-features-1.25.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.4.
* {ServerlessProductName} now uses Knative Eventing 1.4.
* {ServerlessProductName} now uses Kourier 1.4.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.4.
* {ServerlessProductName} now uses Knative Kafka 1.4.
* The `kn func` CLI plug-in now uses `func` 1.7.0.

> * Integrated development environment (IDE) plug-ins for creating and deploying functions are now available for link:https://github.com/redhat-developer/vscode-knative[Visual Studio Code] and link:https://github.com/redhat-developer/intellij-knative[IntelliJ].

Author: It's either that, or for another reason, but I got a green light on including these.

* Knative Kafka broker is now GA. Knative Kafka broker is a highly performant implementation of the Knative broker API, directly targeting Apache Kafka.
+

> It is recommended to not use the MT-Channel-Broker, but the Knative Kafka Broker instead.

Reviewer: Should "Knative Kafka Broker" be "Knative Kafka broker"? In other references in this PR and the docs, the "broker" aspect of "Kafka broker" is not capitalised.

> * Knative Kafka sink is now GA. A `KafkaSink` takes a `CloudEvent` and sends it to an Apache Kafka Topic. Events can be specified in either structured or binary content modes.

Reviewer: Should "Topic" be in back-ticks or not capitalised?

// Module included in the following assemblies
//
// * serverless/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-25-0_{context}"]
= Release notes for Red Hat {ServerlessProductName} 1.25.0

{ServerlessProductName} 1.25.0 is now available. New features, changes, and known issues that pertain to {ServerlessProductName} on {product-title} are included in this topic.

[id="new-features-1.25.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.4.
* {ServerlessProductName} now uses Knative Eventing 1.4.
* {ServerlessProductName} now uses Kourier 1.4.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.4.
* {ServerlessProductName} now uses Knative Kafka 1.4.

> * The `kn func` CLI plug-in now uses `func` FIXME.

Reviewer 2: func 1.7.0

> * The `kn func` CLI plug-in now uses `func` 1.7.0.

Reviewer 3: ```suggestion
* The `kn func` CLI plug-in now uses Knative Functions 1.7.0.
```

Author: @abrennan89 I've changed this per Lance's suggestion. Seems like GA is a good time to `s/func/Knative Functions/`. Please let me know if I should revert back to "`func`".

Reviewer 4: lgtm

* {FunctionsProductName} is now generally available (GA). Only the Quarkus runtime is generally available. The other provided runtimes are available as a Technology Preview (TP). See xref:../serverless/functions/serverless-functions-getting-started.adoc#serverless-functions-getting-started[Getting started with functions].
* Integrated development environments (IDE) plug-ins for creating and deploying functions are now available for link:https://github.com/redhat-developer/vscode-knative[Visual Studio Code] and link:https://github.com/redhat-developer/intellij-knative[IntelliJ].

> * Knative Kafka Broker is now generally available (GA). Knative Kafka Broker is a highly performant implementation of the Knative Broker API, directly targeting Apache Kafka.

Reviewer 5: ```suggestion
* Knative Kafka Broker is now generally available (GA). Knative Kafka Broker is a highly performant implementation of the Knative Broker API, directly targeting Apache Kafka.
* Knative Kafka Sink is now generally available (GA).
```

> * Knative Kafka Broker is now generally available (GA). Knative Kafka Broker is a highly performant implementation of the Knative Broker API, directly targeting Apache Kafka.

Reviewer 5: As part of the GA for Broker and KafkaSink, did we remove the TP notes in docs, like at page https://docs.openshift.com/container-platform/4.7/serverless/admin_guide/serverless-kafka-admin.html 
```
Kafka broker (Technology Preview)

Kafka sink (Technology Preview)
```
or at this section https://docs.openshift.com/container-platform/4.7/serverless/develop/serverless-using-brokers.html#serverless-using-brokers-kafka_serverless-using-brokers

Reviewer 5: (there might be other instances I haven't checked all pages)

+
It is recommended to not use the MT-Channel-Broker, but the Knative Kafka Broker instead.

> * Enabling TLS for internal traffic is now available as a Technology preview. See the "Global configuration" section (FIXME link) for details.

Reviewer 6: I think we can(or should?) add the `TLS for internal traffic` to the feature tables in addition to the here release note. What do you think?

Author: @nak3 Done! Good catch.

Reviewer 7: @rh-max Nit. Should it be `Technology Preview` instead of `Technology preview` ? We use capitals a few paragraphs above so let's unify it.

Author: @mgencur Fixed the nit, thanks.

// the link is https://50404--docspreview.netlify.app/openshift-enterprise/latest/serverless/security/serverless-config-tls.html

[id="fixed-issues-1.25.0_{context}"]
== Fixed issues

* Previously, Knative Serving had an issue where the readiness probe failed if the container was restarted after a liveness probe fail. This issue has been fixed.

[id="known-issues-1.25.0_{context}"]
== Known issues

* The Federal Information Processing Standards (FIPS) mode is disabled for Kafka broker, Kafka source, and Kafka sink.

> * `SinkBinding` is not supported for bFIXME yFIXME oFIXME (BYO) revision names.

Reviewer 7: There are still a few "fixme" on this line and in other places as well. Are you expecting some info for these or you will change this later?

Author: @mgencur Thanks. The only remaining FIXME now is awaiting merge of another PR to create the link to another section.

== New features

* {ServerlessProductName} now uses Knative Serving 1.4.
* {ServerlessProductName} now uses Knative Eventing 1.4.
* {ServerlessProductName} now uses Kourier 1.4.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.4.
* {ServerlessProductName} now uses Knative Kafka 1.4.
* The `kn func` CLI plug-in now uses `func` 1.7.0.

* {FunctionsProductName} is now generally available (GA). Only the Quarkus runtime is generally available. The other provided runtimes are available as a Technology Preview (TP). See xref:../serverless/functions/serverless-functions-getting-started.adoc#serverless-functions-getting-started[Getting started with functions].
* Integrated development environments (IDE) plug-ins for creating and deploying functions are now available for link:https://github.com/redhat-developer/vscode-knative[Visual Studio Code] and link:https://github.com/redhat-developer/intellij-knative[IntelliJ].
* Knative Kafka Broker is now generally available (GA). Knative Kafka Broker is a highly performant implementation of the Knative Broker API, directly targeting Apache Kafka.
+
It is recommended to not use the MT-Channel-Broker, but the Knative Kafka Broker instead.

> * Knative Kafka Sink is now generally available (GA). Knative Kafka Sink is an event sink that you can use if Kafka is enabled in your cluster.

Reviewer 5: ```suggestion
* Knative Kafka Sink is now generally available (GA). A KafkaSink takes a CloudEvent and sends it to an Apache Kafka Topic. Events can be specified in either Structured or Binary mode.
```

> * Knative Kafka Sink is now generally available (GA). A `KafkaSink` takes a `CloudEvent` and sends it to an Apache Kafka Topic. Events can be specified in either Structured or Binary mode.

Reviewer 8: Not sure but possibly "structured" and "binary" do not need capitalization

* Enabling TLS for internal traffic is now available as a Technology Preview. See the "Global configuration" section (FIXME link once https://github.com/openshift/openshift-docs/pull/50404 is merged) for details.
// the link is https://50404--docspreview.netlify.app/openshift-enterprise/latest/serverless/security/serverless-config-tls.html

[id="fixed-issues-1.25.0_{context}"]
== Fixed issues

* Previously, Knative Serving had an issue where the readiness probe failed if the container was restarted after a liveness probe fail. This issue has been fixed.

[id="known-issues-1.25.0_{context}"]
== Known issues

> * The Federal Information Processing Standards (FIPS) mode is disabled for Kafka broker, Kafka source, and Kafka sink.

Reviewer 8: Kafka broker, Kafka source, and Kafka sink are capitalized in section above - pointing out for consistency

== Known issues

* The Federal Information Processing Standards (FIPS) mode is disabled for Kafka broker, Kafka source, and Kafka sink.

> * The `SinkBinding` object does not support custom revision names for services.

Reviewer 9: lgtm
